From: Christopher Harvey <charvey@matrox.com>
Date: Wed Jun 5 11:29:57 2013 -0400
Subject: drm/mgag200: Don't do full cleanup if mgag200_device_init fails
Patch-mainline: Upstream/v3.11-rc1
Git-commit: 279119776d98305af08466c6cc9b7869c57c706f

References: FATE#317582
Signed-off-by: Egbert Eich <eich@suse.com>

Running mgag200_driver_unload when the driver init fails early on
causes functions like drm_mode_config_cleanup to be called. The
problem is, drm_mode_config_cleanup crashes because the corresponding
init hasn't happend yet. There really isn't anything to cleanup after
mgag200_device_init, so we can just pass the error code upwards.

Acked-by: Julia Lemire <jlemire@matrox.com>
Signed-off-by: Christopher Harvey <charvey@matrox.com>
Acked-by: Paul Menzel <paulepanter@users.sourceforge.net>
Signed-off-by: Dave Airlie <airlied@gmail.com>
---
 drivers/gpu/drm/mgag200/mgag200_main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/gpu/drm/mgag200/mgag200_main.c
+++ b/drivers/gpu/drm/mgag200/mgag200_main.c
@@ -232,7 +232,7 @@ int mgag200_driver_load(struct drm_devic
 	r = mgag200_device_init(dev, flags);
 	if (r) {
 		dev_err(&dev->pdev->dev, "Fatal error during GPU init: %d\n", r);
-		goto out;
+		return r;
 	}
 	r = mgag200_mm_init(mdev);
 	if (r)
